### PR TITLE
LESQ-830: Plan a lesson page

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: {owa_music_year}

Pulls together the majority of the new plan a lesson page utilising the components built in the Oak Component library

- Adds BlogAndWebinar list component
- LessonPlanningBlog component
- amends getAndMergeWebinarsAndBlogs function
- Adds corresponding test files

## Issue(s)

Fixes #LESQ-830

## How to test

1. Go to https://deploy-preview-2420--oak-web-application.netlify.thenational.academy/lesson-planning-new

## Screenshots

How it should now look:

https://github.com/oaknational/Oak-Web-Application/assets/91190841/7ef785bc-14be-4a21-b9ba-1b6ae31fc9ed

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/SharedComponents/PortableText/PortableText.tsx [hunk 1]
```diff
@@ -149,6 +154,13 @@ const BodyP = styled(OakP)`
 
 export const basePortableTextComponents: PortableTextComponents = {
   block: {
+    sectionHeading: (props) => {
```
Comment (JBR90) [inline src/components/SharedComponents/PortableText/PortableText.tsx:157]: Will this effect any styling in other parts of the site, blogs etc ?

**Review by JBR90 — CHANGES_REQUESTED** [review]
Looks good, one small style thing I noticed , on mobile the contents hero and blog section are not quite aligned.

image

**Review by jack-skerrett-oak — APPROVED** [review]
I think it is probably worth while to get this merged (once other's comments are handled) and then I can spend some time testing out all the Sanity content block functionality and create tickets for general page feedback rather than putting it all onto this PR.

File: src/components/SharedComponents/PortableText/PortableText.tsx (hunk 1 above)
Comment (k-huggs) [reply src/components/SharedComponents/PortableText/PortableText.tsx]: Removed

**Comment by k-huggs** [comment]
> Looks good, one small style thing I noticed , on mobile the contents hero and blog section are not quite aligned.
> 
> image

Fixed on component lib PR ✅

File: src/components/GenericPagesComponents/LandingPageSignUpForm/LandingPageSignUpForm.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ export const LandingPageSignUpForm: FC<{ formTitle: string }> = ({
   const descriptionId = `${id}-newsletter-form-description`;
   return (
     <Card
-      $ml={[0, 48]}
+      // $ml={[0, 48]}
```
Comment (JBR90) [inline src/components/GenericPagesComponents/LandingPageSignUpForm/LandingPageSignUpForm.tsx:18]: Should this be commented out ?

File: src/components/GenericPagesComponents/PostImageWithAltText/PostImageWithAltText.tsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ const PostImageWithAltText = (
     return null;
   }
 
-  return <CMSImage image={props.value} $mt={80} $mb={64} />;
+  return <CMSImage image={props.value} $mt={24} $mb={24} />;
```
Comment (JBR90) [inline src/components/GenericPagesComponents/PostImageWithAltText/PostImageWithAltText.tsx:14]: This is change how blogs will look, is it a change we want ?

**Review by JBR90 — APPROVED** [review]
Approved with a couple of comments re changes to blog styling

File: src/components/GenericPagesComponents/PostImageWithAltText/PostImageWithAltText.tsx (hunk 3 above)
Comment (k-huggs) [reply src/components/GenericPagesComponents/PostImageWithAltText/PostImageWithAltText.tsx:14]: I'll double check with Monika, but my feeling is that all the blogs should have the same styling for consistency

File: src/components/GenericPagesComponents/LandingPageSignUpForm/LandingPageSignUpForm.tsx (hunk 2 above)
Comment (k-huggs) [reply src/components/GenericPagesComponents/LandingPageSignUpForm/LandingPageSignUpForm.tsx]: removed ✅

**Review by Monika-Koziol — APPROVED** [review]

**Pull request merged**